bin/lib/Logos/Generator.pm

```diff
@@ -40,9 +42,20 @@ sub use {
 		unshift @INC, $2;
 	}
 	$GeneratorPackage = "Logos::Generator::".$generatorName;
-	::fileError(-1, "I can't find the $generatorName Generator!") if(!can_load(modules => {
-				$GeneratorPackage."::Generator" => undef
-			}));
+	::fileError(-1, "I can't find the '$generatorName' Generator!") if (!can_load(modules => {
+		$GeneratorPackage . "::Generator" => undef
+	}));
+
+	# Guard against case insensitive filesystems finding the module but not being able to load it
+	my @availableGeneratorPaths = glob(dirname(__FILE__) . "/Generator/*");
+	my @availableGenerators = map {basename($_)} @availableGeneratorPaths;
+	my $generatorDirectoryExists = any {$_ eq $generatorName} @availableGenerators;
+
+	if ($generatorDirectoryExists != 1) {
+		my %generatorNames = map {lc($_) => $_} @availableGenerators;
+		my $possibleGeneratorName = $generatorNames{lc($generatorName)};
+		::fileError(-1, "I can't find the '$generatorName' Generator, did you mean '$possibleGeneratorName'?");
```

Comment:
For case-sensitive OS support:
- Move this new code block above the existing `$GeneratorPackage` block and fall back to the existing fileError if we have no name suggestion.

```suggestion
		if ($possibleGeneratorName) {
		       ::fileError(-1, "I can't find the '$generatorName' Generator, did you mean '$possibleGeneratorName'?");
		}
```